aws-cpp-sdk-core/source/client/AWSClient.cpp

```diff
@@ -73,6 +73,8 @@ static CoreErrors GuessBodylessErrorType(Aws::Http::HttpResponseCode responseCod
         return CoreErrors::ACCESS_DENIED;
     case HttpResponseCode::NOT_FOUND:
         return CoreErrors::RESOURCE_NOT_FOUND;
+    case HttpResponseCode::BAD_REQUEST:
```

Comment:
This is the main change of this PR, everything else is plumbing to make the BAD_REQUEST error code available in every error class.